gcp-cloud-run/50-cloud-run.tf

```diff
@@ -1,122 +1,121 @@
-        }
-        env {
-          name  = "PVAULT_DB_NAME"
-          value = var.cloudsql_name
-        }
-        env {
-          name  = "PVAULT_DB_USER"
-          value = var.cloudsql_username
-        }
-        env {
-          name  = "PVAULT_SERVICE_LICENSE"
-          value = var.pvault_service_license
-        }
-        env {
-          name  = "PVAULT_SERVICE_TIMEOUT_SECONDS"
-          value = "27"
-        }
-        env {
-          name  = "PVAULT_KMS_URI"
-          value = "gcp-kms://${google_kms_crypto_key.vault-encryption-key.key_ring}/cryptoKeys/${google_kms_crypto_key.vault-encryption-key.name}"
-        }
-        env {
-          name  = "PVAULT_DEVMODE"
-          value = var.pvault_devmode ? "1" : "0"
+    service_account = google_service_account.pvault-server-sa.email
+
+    timeout                          = "28s"
+    max_instance_request_concurrency = var.cloud_run_scaling.max_instance_request_concurrency
+    scaling {
+      min_instance_count = var.cloud_run_scaling.min_instance_count
+      max_instance_count = var.cloud_run_scaling.max_instance_count
+    }
+
+    annotations = {
+      "piiano.io/deployment_id" = var.deployment_id
+    }
+
+    containers {
+      image = "${var.pvault_repository}:${var.pvault_tag}"
+
+      dynamic "env" {
+        for_each = local.pvault_env
+        content {
+          name  = env.key
+          value = env.value
         }
-        env {
-          name = "PVAULT_DB_PASSWORD"
-          value_from {
-            secret_key_ref {
-              name = google_secret_manager_secret.db_password_secret.secret_id
-              key  = "latest"
-            }
+      }
+      env {
+        name = "PVAULT_DB_PASSWORD"
+        value_source {
+          secret_key_ref {
+            secret  = google_secret_manager_secret.db_password_secret.secret_id
+            version = "latest"
           }
         }
-        env {
-          name = "PVAULT_SERVICE_ADMIN_API_KEY"
-          value_from {
-            secret_key_ref {
-              name = google_secret_manager_secret.admin_api_key.secret_id
-              key  = "latest"
-            }
+      }
+      env {
+        name = "PVAULT_SERVICE_ADMIN_API_KEY"
+        value_source {
+          secret_key_ref {
+            secret  = google_secret_manager_secret.admin_api_key.secret_id
+            version = "latest"
           }
         }
-        ports {
-          container_port = 8123
+      }
+
+      ports {
+        container_port = 8123
+      }
+
+      startup_probe {
+        initial_delay_seconds = 0
+        timeout_seconds       = 1
+        period_seconds        = 1
+        failure_threshold     = 60
+        tcp_socket {}
+      }
+      liveness_probe {
+        initial_delay_seconds = 0
+        timeout_seconds       = 5
+        period_seconds        = 10
+        failure_threshold     = 3
+        http_get {
+          path = "/api/pvlt/1.0/ctl/info/health"
```

Comment:
Let's change that to [data](https://docs.piiano.com/api/operations/data-health) because the control healthcheck makes a DB transaction, so data is lighter on the DB.

Reply:
Thanks, wasn't sure how deep the endpoint was checking, so left it out of startupProbe.

Added the data path into startup probe too.